### ISSUE TITLE
Pyenv and UV virtualenv directories coexistence 

### ISSUE THREAD

**Issue by WSF-SEO-AM** [astral-sh/uv#8072]
I've a couple of projects of mine that have been started with pyenv only.. I'm now attracted by the idea of using UV as a factotum, but I'm unsure whether the existing pyenv virtual environment can be designated as a location for UV too.

See below a preliminary attempt of setting the environment path:
- name of pyenv local, the python executable was correctly set, but project based .venv directory and site packages was created
- path of the global shim, no .venv as far as I've seen, but a /bin/bin and a bin/lib were created. Unable to understand how to activate the UV at that point, unsure whether site packages works have been dispatched
- grabbed the full venvs path generated by pyenv. Locations were just fine as much as I can say, but how one can activate the UV environment was not clear

Is this concurrent setup something for which support should be expected, or UV is meant to run only on its own?

**Comment by WSF-SEO-AM** [comment]
So, it seems like this co-existince is possible so long that pyenv creates the virtual environment and the uv venv is not used, rather the enviroment is activated and deactived via pyenv itself.
By doing this it is possble to leverage on uv managing the pyproject.toml file as well as the pip command and it's increased speed.

**Comment by zanieb** [comment]
Hi! Did you see the [`UV_PROJECT_ENVIRONMENT`](https://docs.astral.sh/uv/concepts/projects/#configuring-the-project-environment-path) setting?

**Comment by WSF-SEO-AM** [comment]
Nope. I'm currently discovering the bits and bobs. Can you suggest how this is used?
Is it a one off when the venv is invoked for the first time?
Or, like the TLS setting one, it has to be specified all the times?

**Comment by zanieb** [comment]
It has to be specified at all times. In your case, I'd set it up so it automatically changes per directory, e.g., with `direnv` or as a variable in the `pyenv` environment?

**Comment by WSF-SEO-AM** [comment]
So that I understand ... all the times I invoke a source venv activate? Do I need to specify it there all the times?
Is any place where I can hardcode the path on a per project basis?

**Comment by WSF-SEO-AM** [comment]
@zanieb sorry to chase. Can u please help me here providing more guidance on how to setup uv with a conf file if necessary that specify a variable it can read without the need to make very much complicate command lines?

**Comment by eepstein** [comment]
check out https://direnv.net/  you add a single completion script to your shell's rc file and then can add a .envrc file to any directory and when you're in that directory the .envrc can set env vars including by sourcing a .venv/bin/activate file, etc. or by just setting the env var in the .envrc file.  Another option would be to edit the pyenv-virtualenv activate file in the virtualenv's bin folder.
